client/scripts/controllers/server/search.ts

```diff
@@ -0,0 +1,297 @@
+        return response.result;
+    } catch (e) {
+        console.error(e);
+        return [];
+    }
+  }
+
+  private searchComments = async (
+    searchTerm: string,
+    params: SearchParams
+  ) => {
+    const { resultSize, chainScope, communityScope, sort } = params;
+    try {
+        const response = await $.get(`${app.serverUrl()}/searchComments`, {
+        chain: chainScope,
+        community: communityScope,
+        cutoff_date: null, // cutoffDate.toISOString(),
+        search: searchTerm,
+        results_size: resultSize,
+        sort
+        });
+        if (response.status !== 'Success') {
+        throw new Error(`Got unsuccessful status: ${response.status}`);
+        }
+        return response.result;
+    } catch (e) {
+        console.error(e);
+        return [];
+    }
+  }
+
+  public searchThreadTitles = async (
+    searchTerm: string,
+    params: SearchParams
+  ): Promise<OffchainThread[]> => {
+    const { resultSize, chainScope, communityScope } = params;
+    try {
+      const response = await $.get(`${app.serverUrl()}/searchDiscussions`, {
+        chain: chainScope,
+        community: communityScope,
+        search: searchTerm,
+        results_size: resultSize,
+        thread_title_only: true,
+      });
+      if (response.status !== 'Success') {
+        throw new Error(`Got unsuccessful status: ${response.status}`);
+      }
+      return response.result.map((rawThread) => {
+        return modelFromServer(rawThread);
+      });
+    } catch (e) {
+      console.error(e);
+      return [];
+    }
+  };
+
+  public searchMentionableAddresses = async (
+    searchTerm: string,
+    params: SearchParams,
+    order?: string[]
+  ) => {
+    const { resultSize, communityScope, chainScope } = params;
+    try {
+      const response = await $.get(`${app.serverUrl()}/bulkAddresses`, {
+        chain: chainScope,
+        community: communityScope,
+        limit: resultSize,
+        searchTerm,
+        order,
+      });
+      if (response.status !== 'Success') {
+        throw new Error(`Got unsuccessful status: ${response.status}`);
+      }
+      return response.result;
+    } catch (e) {
+      console.error(e);
+      return [];
+    }
+  };
+
+  private sortResults = (a, b) => {
+    // TODO: Token-sorting approach
+    // Some users are not verified; we give them a default date of 1900
+    const aCreatedAt = moment(
+      a.created_at || a.createdAt || a.verified || '1900-01-01T:00:00:00Z'
+    );
+    const bCreatedAt = moment(
+      b.created_at || b.createdAt || b.verified || '1900-01-01T:00:00:00Z'
+    );
+    return bCreatedAt.diff(aCreatedAt);
+  };
+
+  public getHistory() {
+    const rawHistory = JSON.parse(localStorage.getItem(SEARCH_HISTORY_KEY)) || []
+    const history = []
+    for(let i = 0; i < rawHistory.length; i++){
+      history.push(SearchQuery.fromEncodedString(rawHistory[i]))
+    }
+    return history
+  }
+
+  public addToHistory(query: SearchQuery) {
```

Comment:
Can we keep invalid queries (too short etc) out of the search history?

Reply:
Good catch! Fixed